lib/src/piece/bound.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import '../back_end/code_writer.dart';
+import '../constants.dart';
+import 'piece.dart';
+
+/// A piece for a type parameter and its bound.
+///
+/// Handles not splitting before `extends` if we can split inside the bound's
+/// own type arguments, or splitting before `extends` if that isn't enough to
+/// get it to fit.
+final class BoundPiece extends Piece {
```

Comment:
[nit] since "bind" is a verb we can to to pieces this name might be confusing. Consider `TypeBoundPiece` or similar to make it more clearn.

Reply:
Oh, good point.